### ISSUE TITLE
make_axes_locatable unexpectedly rescales inset_axes

### ISSUE THREAD

**Issue by sappelhoff** [matplotlib/matplotlib#18972]
### Bug report

make_axes_locatable behaves unexpectedly with inset_axes, by scaling the inset_axes to a different size.

there was also a short discussion about this on gitter: [November 18, 2020 3:46 PM](https://gitter.im/matplotlib/matplotlib?at=5fb533b1dc70b5159a1d15b9)

**Code for reproduction**

```python
import matplotlib.pyplot as plt
from mpl_toolkits.axes_grid1.inset_locator import inset_axes
from mpl_toolkits.axes_grid1 import make_axes_locatable

fig, ax = plt.subplots(1, 1, figsize=(10, 8))

ax1 = inset_axes(
    ax,
    width="60%",
    height="60%",
    loc="upper left",
    bbox_to_anchor=(0, 0, 1, 1),
    bbox_transform=ax.transAxes,
)

ax2 = inset_axes(
    ax,
    width="30%",
    height="25%",
    loc="upper right",
    bbox_to_anchor=(0, 0, 1, 1),
    bbox_transform=ax.transAxes,
)

```

produces as expected:

![image](https://user-images.githubusercontent.com/9084751/99534024-0e3ba800-29a7-11eb-9a17-73ca19795b27.png)

Now let's assume I want to add a new axes object to add a colorbar to ax2:

```Python
divider = make_axes_locatable(ax2)
```

However, I can't even proceed to create new axes based on `divider`, because the figure now looks like this (unexpectedly)

![image](https://user-images.githubusercontent.com/9084751/99534076-257a9580-29a7-11eb-8079-29598bf44a91.png)

**Expected outcome**

I expected my ax2 to remain where it was, and would then have proceeded to append a new axes right next to it, using something like:

```Python
divider = make_axes_locatable(ax2)
cax = divider.append_axes("right", size="15%", pad=0.05)
```

**Matplotlib version**

  * Operating system: Ubuntu 18.04
  * Matplotlib version: 3.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): `module://ipykernel.pylab.backend_inline`
  * Python version: 3.7
  * Jupyter version (if applicable):
  * Other libraries:

matplotlib installed from default conda channel

**Comment by jklymak** [comment]
As noted on gitter, I would really recommend against laying out your axes as inset_axes.  If you really want manual control, simply lay the axes out by hand.  If you want the idea of subfigures, that is a new feature that will be available in 3.4, or on master.

**Comment by anntzer** [comment]
From a quick look, it would seem reasonable for AxesDivider (which is used to implement make_axes_locatable) to record whether the axes already had a preexisting locator set, and if so use it as basis for division.  Unfortunately, things are made more complicated (I think) by the fact that the AxesDivider can't easily actually call the locator to figure out the axes position, because the locator also needs the renderer as argument but the renderer isn't forwarded to the relevant call sites for the AxesDivider.
